frontend/src/scenes/session-recordings/apm/playerInspector/ItemPerformanceEvent.tsx

```diff
@@ -107,14 +107,20 @@ function StartedAt({ item }: { item: PerformanceEvent }): JSX.Element | null {
     ) : null
 }
 
-function DurationDescription({ item }: { item: PerformanceEvent }): JSX.Element | null {
+function durationMillisecondsFrom(item: PerformanceEvent): number | null {
     let duration = item.duration
     if (duration === undefined && item.end_time !== undefined && item.start_time !== undefined) {
         duration = item.end_time - item.start_time
     }
-    if (duration === undefined) {
+    return duration ?? null
+}
+
+function DurationDescription({ item }: { item: PerformanceEvent }): JSX.Element | null {
+    const duration = durationMillisecondsFrom(item)
+    if (duration === null) {
         return null
     }
```

Comment:
Only ever a personal nit but I always find it easier to do this early return outside of a component. It makes things clearer elsewhere in the code that a component will be rendered. Also makes the component more flexible

Reply:
in this case we're building a sentence and do want this to be null if there's no duration to render
and the sentence code would be super hard to parse imo

i remain torn on this pattern generally 🙈